src/definitions/device_engagement.rs

```diff
@@ -273,45 +315,51 @@ impl TryFrom<CborValue> for DeviceRetrievalMethod {
     }
 }
 
-impl From<DeviceRetrievalMethod> for CborValue {
+impl From<DeviceRetrievalMethod> for ciborium::Value {
     fn from(drm: DeviceRetrievalMethod) -> Self {
         let transport_type = drm.transport_type().into();
         let version = drm.version().into();
         let retrieval_method = match drm {
-            DeviceRetrievalMethod::NFC(opts) => opts.into(),
-            DeviceRetrievalMethod::BLE(opts) => opts.into(),
-            DeviceRetrievalMethod::WIFI(opts) => opts.into(),
+            DeviceRetrievalMethod::NFC(opts) => cbor::into_value(opts).unwrap(),
+            DeviceRetrievalMethod::BLE(opts) => cbor::into_value(opts).unwrap(),
+            DeviceRetrievalMethod::WIFI(opts) => cbor::into_value(opts).unwrap(),
```

Comment:
Again I think this change is unnecessary since we have `From<ciborium::Value>` implemented for all of these types.

Reply:
changed